### DISCUSSION TITLE
read data froom tcp stream on demand

### DISCUSSION THREAD

**Discussion by bokunodev** [libuv/libuv#4397] [General]
is it possible to read data from a stream on demand?
`uv_read_start` ask for a callback that would be called whenever something is available on the stream. that means, the loop is in charge of "when" to read something.
does blocking the callback with `uv_cond_t` would also block the event loop?

i want to spin up a thread and having that thread tobe incharge of when to read something.

**Comment by bnoordhuis** [comment]
Combine uv_read_start with uv_read_stop? If that's not what you mean, then please clarify your question.

> does blocking the callback with uv_cond_t would also block the event loop?

Yes. Sounds like you want uv_async_t?

**Reply by bokunodev** [reply]
does calling `uv_read_stop` from within the callback is allowed?

**Reply by saghul** [reply]
Yes it is.
